File: Quorum/solve/javascript/QuorumSolver.js
#! /usr/bin/env node

var fs = require("fs");
var path = require("path");
var tc = null;

function init() {
  var gettcHome = process.env.GETTC_HOME || path.join(process.env.HOME, ".gettc");
  var includeDir = path.join(gettcHome, "include", "javascript");
  tc = require(path.join(includeDir, "topcoder"));
}

function main() {
  try {
    var input = fs.readFileSync(process.argv[2], { encoding: "ascii" });
    var reader = new tc.Reader(input);
        var arr = reader.next('int[]'); reader.next();
    var k = reader.next('int');

    var Quorum = require("./Quorum");
    var result = Quorum.count(arr, k);

    var writer = new tc.Writer();
    writer.next(result, "int");
    fs.writeFileSync(process.argv[3], writer.toString(), { encoding: "ascii" });
  } catch (err) {
    console.log(err.toString());
    console.trace();
  }
}

init();
main();

File: Quorum/solve/ruby/QuorumSolver.rb
#! /usr/bin/env ruby

require_relative "Quorum"

require "gettc/types"
include Gettc

def init
  gettc_home = File.expand_path(ENV["GETTC_HOME"] || File.join(ENV["HOME"], ".gettc"))
  $LOAD_PATH << File.join(gettc_home, "include", "ruby")
  require "topcoder"
  include TopCoder
end

def main
  reader = Reader.new(IO.read(ARGV[0]))
  arr = reader.next(TArray.new(TInt))
  reader.next()
  k = reader.next(TInt)

  result = Quorum.new().count(arr, k)
  IO.write(ARGV[1], Writer.new.next(result, TInt).to_s)
end

init
main
